internal/service/controller_service.go

```diff
@@ -302,14 +304,67 @@ func (s *ControllerService) Listen(req *pb.ListenRequest, stream pb.ControllerSe
 		},
 		Reason: "Connect",
 	})
-	if err = s.Status().Patch(ctx, exporter, original); err != nil {
+	if err = s.Client.Status().Patch(ctx, exporter, original); err != nil {
 		logger.Error(err, "unable to update exporter status", "exporter", exporter)
 	}
 
 	<-ctx.Done()
 	return nil
 }
 
+func (s *ControllerService) Status(req *pb.StatusRequest, stream pb.ControllerService_StatusServer) error {
+	ctx := stream.Context()
+	logger := log.FromContext(ctx)
+
+	exporter, err := s.authenticateExporter(ctx)
+	if err != nil {
+		return err
+	}
+
+	watcher, err := s.Client.Watch(ctx, &jumpstarterdevv1alpha1.ExporterList{}, &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector("metadata.name", exporter.Name),
+		Namespace:     exporter.Namespace,
+	})
+	if err != nil {
+		logger.Error(err, "failed to watch exporter")
+		return err
+	}
+
+	defer watcher.Stop()
+	for result := range watcher.ResultChan() {
+		switch result.Type {
+		case watch.Added, watch.Modified, watch.Deleted:
+			exporter = result.Object.(*jumpstarterdevv1alpha1.Exporter)
```

Comment:
Right, when you subscribe to a watcher, you get an initial update with the current state. Perfect.